### ISSUE TITLE
How do i use it or anagram solving

### ISSUE THREAD

**Issue by kenofori** [bzaar/DawgSharp#23]
Hi there,

Great library, and im considering it for commercial user however there are few things i need to clear.
How can i use it for solving anagram ? Lets suppose i have random letters like a,c,x,y and what i want to see how many words can be formed with only these 4 letters what function i use?

Thanks

**Comment by bzaar** [comment]
Quick googling for "anagram solving algo" led me to this nice post: https://hackernoon.com/an-algorithm-for-finding-anagrams-2fe7655de85b

The gist of it is a function that maps a word to a number so that anagrams get the same number. It does so by assigning each letter (A...Z) a prime number and multiplying the primes that make up the word.

The rest is trivial: just find all the words whose f(w) matches f(input).

You can use DawgSharp to store the words and keep your memory footprint low.

**Comment by kenofori** [comment]
@bzaar Thank you for your quick reply,
Sorry, english is not my primary language so i guess i used the wrong word/term.

Let me try to explain what i want to do in more detail

Lets suppose i have given 5-6 random letters ( M, C, A, P, T)

and this is current DAWG structure in memory

C
__|_____________
|                      |
A                    O
_|_____             |
|         |
T[$]    R[$]     M[$]
|
S [$]

In short, DAWG has Cats, Cat, Car and Com word



Now i will construct a word sequence from this letters starting with "C" and "A" = "CA"

Now i need a function which return me all the direct children node
GetDirectChildre("CA");
returns me R[$ and T[$] nodes and now i know there is only one word if i take "R" route/edge.
But T[$] is itself a word and has 2 more child count.

Is there something like that?
Thanks

**Comment by bzaar** [comment]
There is the method [Dawg.MatchPrefix](https://github.com/bzaar/DawgSharp/blob/master/DawgSharp/Dawg.cs#L29) which returns all words that start with a given sequence: dawg.MatchPrefix"CA") will yield CAT, CATS and CAR in your example. Is that what you're after?

**Comment by kenofori** [comment]
@bzaar thank you very much,
i just resumed working on it again. will be asking more questions
